examples/applications/plot_2d_constrained.py

```diff
@@ -0,0 +1,126 @@
+"""
```

Comment:
This script takes significantly longer to converge than all the other ones; 11s vs <1s on all the others. Maybe one of the parameters is not exactly right. Does converge to a good solution.

Reply:
This may well be due to an internal step-size parameter. @Luke-Pratley would likely have a better insight to the code here.